### PR TITLE
feat(storefrontlib): generator of JSON translation files out from TS files

### PR DESCRIPTION
**Pull request by Platonn**
All translations will be defined in TS files and imported in shell app for purposes of development. But we need a script that outputs JSONs (from TS files) in /dist folder - justo to release also JSON files which are ready to use for lazy-loading.

fixes GH-1735

### CODE REVIEW

File: scripts/generate-translations.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const libDist = './dist/storefrontlib/';
+const translationsDist = libDist + 'translations/';
+function createDir(dir) {
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir);
+  }
+}
+const getLangDir = lang => translationsDist + lang + '/';
+const getFileName = (lang, namespace) =>
+  getLangDir(lang) + namespace + '.' + lang + '.json';
```
Comment (marlass) [inline scripts/generate-translations.ts]: Maybe use here template string for concatenation. Should be more readable.

File: scripts/generate-translations.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  console.log(
+    `Cannot generate translations. Directory '${libDist}' does not exist.`
+  );
+} else {
+  // clear translations dist
+  rimraf.sync(translationsDist);
+  createDir(translationsDist);
+
+  // generate files
+  Object.keys(translations).forEach(lang => {
```
Comment (marlass) [inline scripts/generate-translations.ts]: `Object.entries` might be better for these, but I assume that we still use ES6 version of TS?

File: scripts/generate-translations.ts (hunk 1 above)
Comment (dunqan) [reply scripts/generate-translations.ts]: :+1:

File: scripts/generate-translations.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    createDir(getLangDir(lang));
+    Object.keys(translations[lang]).forEach(namespace => {
+      const obj = translations[lang][namespace];
+      const json = JSON.stringify(obj);
```
Comment (marlass) [inline scripts/generate-translations.ts]: Print JSON formatted for better debugging/changes.
```suggestion
      const json = JSON.stringify(obj, null, 2);
```

File: package.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
   },
   "scripts": {
     "build": "ng build storefrontapp --prod",
-    "build:core:lib": "ng build core && ng build storefrontlib",
+    "build:core:lib": "ng build core && ng build storefrontlib && generate:translations",
```
Comment (marlass) [inline package.json]:
```suggestion
    "build:core:lib": "ng build core && ng build storefrontlib && yarn generate:translations",
```

**Pull request merged**